lib/AST/RequirementMachine/RequirementLowering.cpp

```diff
@@ -455,29 +526,149 @@ void swift::rewriting::realizeInheritedRequirements(
       inferRequirements(inheritedType, loc, moduleForInference, result);
     }
 
-    realizeTypeRequirement(type, inheritedType, loc, result);
+    realizeTypeRequirement(type, inheritedType, loc, result, errors);
   }
 }
 
+/// Emit diagnostics for the given \c RequirementErrors.
+///
+/// \param ctx The AST context, which is used to determine whether
+/// requirement machine protocol signatures are enabled. If not,
+/// diagnostics will not be emitted.
+/// \param errors The set of requirement diagnostics to be emitted.
+/// \param allowConcreteGenericParams Whether concrete type parameters
+/// are permitted in the generic signature. If true, diagnostics will
+/// offer fix-its to turn invalid type requirements, e.g. T: Int, into
+/// same-type requirements.
+///
+/// \returns true if any errors were emitted, and false otherwise (including
+/// when only warnings were emitted).
+bool swift::rewriting::diagnoseRequirementErrors(
+    ASTContext &ctx, SmallVectorImpl<RequirementError> &errors,
+    bool allowConcreteGenericParams) {
+  bool diagnosedError = false;
+  if (ctx.LangOpts.RequirementMachineProtocolSignatures !=
```

Comment:
I think this check should happen in the callers, since it depends on RequirementMachineInferredSignatures also?

Reply:
Oh, I think you're right that this is wrong 🙂 it should be that `InferredGenericSignatureRequestRQM` checks `RequirementMachineInferredSignatures` and the other requests for protocol and typealias requirements check `RequirementMachineProtocolSignatures`, right?